gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinDriverIntegrateTest.java

```diff
@@ -1877,4 +1877,67 @@ public void shouldNotHangWhenSameRequestIdIsUsed() throws Exception {
 
         assertEquals(100, result1.get().one().getInt());
     }
+
+    /**
+     * Client created on an initially dead host should fail initially, and recover after the dead host has restarted
+     * @param testClusterClient - boolean flag set to test clustered client if true and sessioned client if false
+     * @throws Exception
+     */
+    public void testShouldFailOnInitiallyDeadHost(final boolean testClusterClient) throws Exception {
+        logger.info("Stopping server.");
+        this.stopServer();
+
+        final Cluster cluster = TestClientFactory.build().create();
+        final Client client = testClusterClient? cluster.connect() : cluster.connect("sessionClient");
+
+        try {
+            // try to re-issue a request now that the server is down
+            logger.info("Verifying driver cannot connect to server.");
+            client.submit("g").all().get(500, TimeUnit.MILLISECONDS);
+            fail("Should throw an exception.");
+        } catch (RuntimeException re) {
+            // Client would have no active connections to the host, hence it would encounter a timeout
+            // trying to find an alive connection to the host.
+            Throwable foo = re.getCause();
```

Comment:
Better variable name please.

Reply:
Oh that isn't even being used so I just deleted it. Not sure why it was there in the first place lol